sdk/helper/ocsp/client.go

```diff
@@ -0,0 +1,1085 @@
+		return nil, errors.New("OCSP Response is nil")
+	}
+	if !isInValidityRange(curTime, ocspRes.NextUpdate) {
+		return &ocspStatus{
+			code: ocspInvalidValidity,
+			err:  fmt.Errorf("invalid validity: producedAt: %v, thisUpdate: %v, nextUpdate: %v", ocspRes.ProducedAt, ocspRes.ThisUpdate, ocspRes.NextUpdate),
+		}, nil
+	}
+	return returnOCSPStatus(ocspRes), nil
+}
+
+func returnOCSPStatus(ocspRes *ocsp.Response) *ocspStatus {
+	switch ocspRes.Status {
+	case ocsp.Good:
+		return &ocspStatus{
+			code: ocspStatusGood,
+			err:  nil,
+		}
+	case ocsp.Revoked:
+		return &ocspStatus{
+			code: ocspStatusRevoked,
+		}
+	case ocsp.Unknown:
+		return &ocspStatus{
+			code: ocspStatusUnknown,
+			err:  errors.New("OCSP status unknown."),
+		}
+	default:
+		return &ocspStatus{
+			code: ocspStatusOthers,
+			err:  fmt.Errorf("OCSP others. %v", ocspRes.Status),
+		}
+	}
+}
+
+// retryOCSP is the second level of retry method if the returned contents are corrupted. It often happens with OCSP
+// serer and retry helps.
+func (c *Client) retryOCSP(
+	ctx context.Context,
+	client clientInterface,
+	req requestFunc,
+	ocspHost *url.URL,
+	headers map[string]string,
+	reqBody []byte,
+	issuer *x509.Certificate) (ocspRes *ocsp.Response, ocspResBytes []byte, ocspS *ocspStatus, err error) {
+
+	origHost := *ocspHost
+	doRequest := func(request *retryablehttp.Request) (*http.Response, error) {
+		if err != nil {
+			return nil, err
+		}
+		if request != nil {
+			request = request.WithContext(ctx)
+			for k, v := range headers {
+				request.Header[k] = append(request.Header[k], v)
+			}
+		}
+		res, err := client.Do(request)
+		if err != nil {
+			return nil, err
+		}
+		c.Logger().Debug("StatusCode from OCSP Server:", "statusCode", res.StatusCode)
+		return res, err
+	}
+
+	ocspHost.Path = ocspHost.Path + "/" + base64.StdEncoding.EncodeToString(reqBody)
+	var res *http.Response
+	request, err := req("GET", ocspHost.String(), nil)
+	if res, err = doRequest(request); err != nil {
+		return nil, nil, nil, err
+	} else {
+		defer res.Body.Close()
+	}
+	if res.StatusCode == http.StatusMethodNotAllowed {
+		request, err = req("POST", origHost.String(), bytes.NewBuffer(reqBody))
+		if res, err = doRequest(request); err != nil {
+			return nil, nil, nil, err
+		} else {
+			defer res.Body.Close()
+		}
+	}
+	if res.StatusCode != http.StatusOK {
+		return nil, nil, nil, fmt.Errorf("HTTP code is not OK. %v: %v", res.StatusCode, res.Status)
+	}
+	ocspResBytes, err = io.ReadAll(res.Body)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	ocspRes, err = ocsp.ParseResponse(ocspResBytes, issuer)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	return ocspRes, ocspResBytes, &ocspStatus{
+		code: ocspSuccess,
+	}, nil
+}
+
+// GetRevocationStatus checks the certificate revocation status for subject using issuer certificate.
+func (c *Client) GetRevocationStatus(ctx context.Context, subject, issuer *x509.Certificate, conf *VerifyConfig) (*ocspStatus, error) {
```

Comment:
Out of curiosity, how are clients supposed to interpret the results from an internal constant type? Both this exported function and `GetAllRevocationStatus` return `*ocspStatus`, which is an type internal to this package. 

I think this is an issue stemming from making this helper public; the snowflake one is internal.